simulariumio/file_converter.py

```diff
@@ -19,24 +21,207 @@
+    ) -> Dict[str, Any]:
+        """
+        Parse spatial data binary block from a .simularium binary file
+        """
+        block_offset = (
+            int(
+                binary_header["block_offsets"][block_index]
+                / BINARY_SETTINGS.BYTES_PER_VALUE
+            )
+            + BINARY_SETTINGS.BLOCK_HEADER_N_VALUES
+        )
+        spatial_data_version = binary_data["int_view"][block_offset]
+        n_frames = binary_data["int_view"][block_offset + 1]
+        current_frame_offset = block_offset + 2 + 2 * n_frames
+        frame_info = binary_data["int_view"][block_offset + 2 : current_frame_offset]
+        frame_lengths = frame_info[1::2]
+        result = {
+            "version": spatial_data_version,
+            "msgType": 1,
+            "bundleStart": 0,
+            "bundleSize": n_frames,
+            "bundleData": [],
+        }
+        for index in range(n_frames):
+            frame_index = binary_data["int_view"][current_frame_offset]
+            if index == 0:
+                result["bundleStart"] = frame_index
+            frame_n_values = int(frame_lengths[index] / BINARY_SETTINGS.BYTES_PER_VALUE)
+            result["bundleData"].append(
+                {
+                    "frameNumber": frame_index,
+                    "time": binary_data["float_view"][current_frame_offset + 1],
+                    "data": list(
+                        binary_data["float_view"][
+                            current_frame_offset
+                            + 3 : current_frame_offset
+                            + frame_n_values
+                        ]
+                    ),
+                }
+            )
+            current_frame_offset += frame_n_values
+        return result
+
+    @staticmethod
+    def _load_binary(input_file: InputFileData) -> Dict[str, Any]:
+        """
+        Load data from the input file in .simularium binary format and update it.
+        """
+        result = {}
+        data = FileConverter._read_binary_file(input_file)
+        binary_header = FileConverter._binary_header(data)
+        # parse blocks
+        for block_index in range(binary_header["n_blocks"]):
+            block_type, block_length = FileConverter._binary_block_header(
+                block_index, binary_header, data
+            )
+            if block_type == 0:
+                result["spatialData"] = FileConverter._binary_block_json(
+                    block_index, binary_header, data
+                )
+            elif block_type == 1:
+                result["trajectoryInfo"] = FileConverter._binary_block_json(
+                    block_index, binary_header, data
+                )
+            elif block_type == 2:
+                result["plotData"] = FileConverter._binary_block_json(
+                    block_index, binary_header, data
+                )
+            elif block_type == 3:
+                result["spatialData"] = FileConverter._binary_block_spatial_data(
```

Comment:
what happens if there is a block of type 0 and a block of type 3 in the same file?  should we do a check in here and error out if that happens?    or things like two plotdata blocks or two trajectoryinfo blocks will always overwrite the result object.  worth noting in a comment at the very least.